### PR TITLE
DO NOT MERGE VACMS-17860: Updates staff phone field to paragraph field

### PR DESCRIPTION
**Pull request by dsasser**
## Description

Relates to #17860 

## Testing done
Unit tests for the phone number extraction, and manual testing and QA for the phone data migration.

## Screenshots
N/A

## QA steps
TBD

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.
- [ ] If there are field changes, front end output has been thoroughly checked.

### Select Team for PR review

- [ ] `CMS Team`
- [x] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [x] `DO NOT MERGE`

### CODE REVIEW

File: .ddev/nginx_full/nginx-site.conf
Comment (jv-agile6) [inline .ddev/nginx_full/nginx-site.conf:1]: default nginx config added for batch process.

File: .github/CODEOWNERS
Comment (jv-agile6) [inline .github/CODEOWNERS:1]: Custom module va_gov_batch

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -50,6 +50,7 @@
 /docroot/modules/custom/va_gov_api                                  @department-of-veterans-affairs/accelerated-publishing
 /docroot/modules/custom/va_gov_backend                              # @department-of-veterans-affairs/platform-cms-drupal-engineers
 /docroot/modules/custom/va_gov_banner                               @department-of-veterans-affairs/public-websites-cms
+/docroot/modules/custom/va_gov_batch                                @department-of-veterans-affairs/public-websites-cms
```
Comment (jv-agile6) [inline .github/CODEOWNERS:53]: Custom module va_gov_batch

File: composer.json
Comment (jv-agile6) [inline composer.json:1]: added codit_batch_operation and giggsey/libphonenumber-for-php libraries

File: config/sync/core.entity_form_display.node.person_profile.default.yml
Comment (jv-agile6) [inline config/sync/core.entity_form_display.node.person_profile.default.yml:1]: Config changes on field.field.node.person_profile.field_telephone. telephone paragraph added telephone field removed

File: config/sync/codit_batch_operations.settings.yml
Comment (jv-agile6) [inline config/sync/codit_batch_operations.settings.yml:1]: Using codit batch operations, enabled va_gov_batch script set user to 1317 and enabled cron zero.

File: config/sync/field.field.node.person_profile.field_telephone.yml
Comment (jv-agile6) [inline config/sync/field.field.node.person_profile.field_telephone.yml:1]: 👀

File: config/sync/views.view.rich_text_field_audit.yml
Comment (jv-agile6) [inline config/sync/views.view.rich_text_field_audit.yml:1]: added fields for phone number and another for extension

File: docroot/modules/custom/va_gov_batch/src/cbo_scripts/MigratePhoneFieldToParagraph.php
Comment (jv-agile6) [inline docroot/modules/custom/va_gov_batch/src/cbo_scripts/MigratePhoneFieldToParagraph.php:1]: va_gov_batch module for cbo_script to use phone field to paragraph migration

File: docroot/modules/custom/va_gov_batch/src/cbo_scripts/MigratePhoneFieldToParagraph.php [hunk 2]
```diff
@@ -0,0 +1,184 @@
...
+
+use Drupal\codit_batch_operations\BatchOperations;
+use Drupal\codit_batch_operations\BatchScriptInterface;
+use Drupal\Core\Entity\EntityStorageException;
+use Drupal\node\Entity\Node;
+use Drupal\node\NodeInterface;
+use Drupal\paragraphs\Entity\Paragraph;
+use Drupal\paragraphs\ParagraphInterface;
+use Drupal\telephone\Plugin\Field\FieldType\TelephoneItem;
+use JetBrains\PhpStorm\ArrayShape;
```
Comment (jv-agile6) [inline docroot/modules/custom/va_gov_batch/src/cbo_scripts/MigratePhoneFieldToParagraph.php:13]: 👀

Comment (jv-agile6) [inline docroot/modules/custom/va_gov_batch/src/cbo_scripts/MigratePhoneFieldToParagraph.php:13]: 👀

File: docroot/modules/custom/va_gov_profile/va_gov_profile.deploy.php
Comment (jv-agile6) [inline docroot/modules/custom/va_gov_profile/va_gov_profile.deploy.php:1]: add MigratePhoneFieldToParagraph to deploy

**Review by jv-agile6 — APPROVED** [review]
Reviewed code additions and subtraction for VACMS-17860: Updates staff phone field to paragraph field #19038. Followed both form and logic with no identifiable defects.

File: .ddev/nginx_full/nginx-site.conf
[the remark this answers is not in the document]
Comment (dsasser) [reply .ddev/nginx_full/nginx-site.conf:1]: > but I believe that it is intended for this to be generated

That is the case initially, but our setup requires the buffers to be bumped up beyond the default in order to accept large POST requests (graphql). This ddev config change has been [vetted with the CMS team](https://dsva.slack.com/archives/CT4GZBM8F/p1724701371261819?thread_ts=1724692145.752969&cid=CT4GZBM8F).

**Review by omahane — APPROVED** [review]
I've been using this code as the base for my own changes, and it's been working well without issue.

**Review by timcosgrove — CHANGES_REQUESTED** [review]
Just a small request to bring the nginx-site.conf in the PR in line with the current generated ddev one; only substantive change is on the list of handled files.

File: .ddev/nginx_full/nginx-site.conf [hunk 3]
```diff
@@ -0,0 +1,107 @@
+# ddev drupal9 config
```
[the remark this answers is not in the document]
Comment (dsasser) [reply .ddev/nginx_full/nginx-site.conf]: Good catch. [Updated](https://github.com/department-of-veterans-affairs/va.gov-cms/pull/19038/commits/97817bfb55a6f1b936afe3ef3c3c89a49e8650b0).

**Review by timcosgrove — APPROVED** [review]

**Pull request merged**